tests/unit/single_table/test_utils.py

```diff
@@ -38,3 +44,336 @@ def test_detect_discrete_columns():
+        'bar': {'bar_dict': 'value_bar_dict'},
+        'tar': ['value_tar_list_0', 'value_tar_list_1'],
+        'fitted': 'value_1',
+        'distribution': 'value_2',
+        'type': 'value_3'
+    }
+    result = flatten_dict(nested)
+
+    # Asserts
+    expected = {
+        'foo': 'value',
+        'bar__bar_dict': 'value_bar_dict',
+        'tar__0': 'value_tar_list_0',
+        'tar__1': 'value_tar_list_1'
+    }
+    assert result == expected
+
+
+def test_flatten_dict_with_prefix():
+    """Test get flatten dict with some result."""
+    # Run
+    nested = {
+        'foo': 'value',
+        'bar': {'bar_dict': 'value_bar_dict'},
+        'tar': ['value_tar_list_0', 'value_tar_list_1'],
+        'fitted': 'value_1',
+        'distribution': 'value_2',
+        'type': 'value_3'
+    }
+    result = flatten_dict(nested, prefix='test')
+
+    # Asserts
+    expected = {
+        'test__foo': 'value',
+        'test__bar__bar_dict': 'value_bar_dict',
+        'test__tar__0': 'value_tar_list_0',
+        'test__tar__1': 'value_tar_list_1'
+    }
+    assert result == expected
+
+
+def test__key_order():
+    """Test key order."""
+    # Run
+    key_value = ['foo__0__1']
+    result = _key_order(key_value)
+
+    # Asserts
+    assert result == ['foo', 0, 1]
+
+
+def test_unflatten_dict_raises_error_row_index():
+    """Test unflatten dict raises error row_index."""
+    # Run
+    flat = {
+        'foo__0__1': 'some value'
+    }
+
+    err_msg = 'There was an error unflattening the extension.'
+    with pytest.raises(ValueError, match=err_msg):
+        unflatten_dict(flat)
+
+
+def test_unflatten_dict_raises_error_column_index():
+    """Test unflatten dict raises error column_index."""
+    # Run
+    flat = {
+        'foo__1__0': 'some value'
+    }
+
+    err_msg = 'There was an error unflattening the extension.'
+    with pytest.raises(ValueError, match=err_msg):
+        unflatten_dict(flat)
+
+
+def test_unflatten_dict():
+    """Test unflatten_dict."""
+    # Run
+    flat = {
+        'foo__0__foo': 'foo value',
+        'bar__0__0': 'bar value',
+        'tar': 'tar value'
+    }
+    result = unflatten_dict(flat)
+
+    # Asserts
+    expected = {
+        'foo': {0: {'foo': 'foo value'}},
+        'bar': [['bar value']],
+        'tar': 'tar value',
+    }
+    assert result == expected
+
+
+def test_handle_sampling_error():
+    """Test the ``handle_sampling_error`` function.
+
+    Expect that the error is raised at the end of the function.
+
+    Input:
+        - True
+        - a temp file
+        - the sampling error
+
+    Side Effects:
+        - the error is raised.
+    """
+    # Setup
+    error_msg = (
+        'Error: Sampling terminated. Partial results are stored in a temporary file: test.csv. '
+        'This file will be overridden the next time you sample. Please rename the file if you '
+        'wish to save these results.'
+        '\n'
+        'Test error'
+    )
+
+    # Run and assert
```

Comment:
For all of these errors, I think we usually put `error_msg` in the `Run and Assert` section (and Assert is usually capitalized). I don't think you should bother changing it here, since it doesn't really matter, just noting so we are consistent in the future.